src/ReactiveUI.Blazor/ReactiveComponentBase.cs

```diff
@@ -87,15 +88,16 @@ protected override void OnAfterRender(bool firstRender)
             {
                 // The following subscriptions are here because if they are done in OnInitialized, they conflict with certain JavaScript frameworks.
                 var viewModelChanged =
-                    this.WhenAnyValue(x => x.ViewModel)
-                        .Where(x => x != null)
-                        .Publish()
-                        .RefCount();
+                    this.WhenAnyValue(x => x.ViewModel);
 
```

Comment:
Both need Where(x != null). Avoid the Skip below. 

Add RefCount(2) and that should avoid the two observables being made.